dataprofiler/data_readers/csv_data.py

```diff
@@ -104,26 +106,145 @@ def _check_and_return_options(self, options):
                                  'header (0 based index)')
         return options
 
+    
     @staticmethod
-    def _determine_delimiter_of_str(data_as_str):
+    def _guess_delimiter_and_quotechar(data_as_str, quotechar=None,
+                                       preferred=[',', '\t'], omitted=['"', "'"]):
         """
         Automatically checks for what delimiter exists in a text document.
 
         :param data_as_str: Single string containing rows (lines separated by "\n")
         :type data_as_str: str
-        :return: Delimiter, if none can be found None is returned
+        :param preferred: Delimiters that will be weighted and selected if possible
+        :type preferred: list
+        :param omitted: Delimiters that will be omitted from selection
+        :type omitted: list
+        :param quotechar: Character used for quotes
+        :type quotechar: str
+        :return: delimiter, if none can be found None is returned
+        :rtype: str or None
+        :return: quotechar, if none can be found None is returned
         :rtype: str or None
         """
-        sniffer = csv.Sniffer()
-        sniffer.preferred = [',', '\t', ';']  # removes ' ', ':' from preferred
-        try:
-            dialect = sniffer.sniff(data_as_str)
-        except csv.Error as exc:
-            return None
-        return dialect.delimiter
+
+        # Detect vocabulary (and count)
+        vocab = {}
+        for i in range(len(data_as_str)):
+            c = data_as_str[i]
+            if c not in vocab:
+                vocab[c] = 0
+            vocab[c] += 1
+        if '\n' in vocab: vocab.pop('\n')
```

Comment:
Not blocking, but previously mentioned this could be using the builtin python counter, might be faster? not sure it matters if we end up limiting char inputs anyways. Ultimately could be cleaner.
```python
from collections import Counter


vocab = Counter(data_as_str)
vocab.pop('\n', None)
``` 
below with other pops could be same.

Reply:
Yeah, had considered. Initially I was going to do a couple other tricks - didn't end up doing it. I'm fine changing it, but I don't think anything will materially change in runtime